### PR TITLE
 update for new authentication APIs

### PR DESCRIPTION
**Pull request by deads2k**
In 1.13, the authenticator APIs changed.  This picks that change back so that we can have a small pull updating them.

/assign @enj 
@openshift/sig-auth

### CODE REVIEW

File: pkg/oauthserver/authenticator/interfaces.go [hunk 1]
```diff
@@ -1,15 +1,14 @@
 package authenticator
 
 import (
-	"k8s.io/apiserver/pkg/authentication/user"
-
 	"github.com/openshift/origin/pkg/oauthserver/api"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
```
Comment (enj) [inline pkg/oauthserver/authenticator/interfaces.go]: nit: move up

File: pkg/oauthserver/authenticator/password/allowanypassword/anyauthpassword.go [hunk 2]
```diff
@@ -33,5 +33,6 @@ func (a alwaysAcceptPasswordAuthenticator) AuthenticatePassword(username, passwo
 
 	identity := authapi.NewDefaultUserIdentityInfo(a.providerName, username)
 
-	return identitymapper.UserFor(a.identityMapper, identity)
+	user, ok, err := identitymapper.UserFor(a.identityMapper, identity)
```
Comment (enj) [inline pkg/oauthserver/authenticator/password/allowanypassword/anyauthpassword.go]: Please update `identitymapper.UserFor` to return `authenticator.Response`

File: pkg/oauthserver/oauth/registry/registry_test.go [hunk 3]
```diff
@@ -5,6 +5,8 @@ import (
 	"net/http/httptest"
 	"testing"
 
+	"k8s.io/apiserver/pkg/authentication/authenticator"
```
Comment (enj) [inline pkg/oauthserver/oauth/registry/registry_test.go]: nit: move down

File: pkg/oauthserver/authenticator/interfaces.go [hunk 4]
```diff
@@ -1,15 +1,14 @@
...
 )
 
 type Assertion interface {
-	AuthenticateAssertion(assertionType, data string) (user.Info, bool, error)
+	AuthenticateAssertion(assertionType, data string) (*authenticator.Response, bool, error)
```
Comment (enj) [inline pkg/oauthserver/authenticator/interfaces.go]: both of these should take a context now I suppose

**Comment by deads2k** [comment]
terraform

/retest

File: pkg/oauthserver/server/login/login.go [hunk 5]
```diff
@@ -186,8 +187,8 @@ func (l *Login) handleLogin(w http.ResponseWriter, req *http.Request) {
 		result = metrics.FailResult
 		return
 	}
-	glog.V(4).Infof(`Login with provider %q succeeded for %q: %#v`, l.provider, username, user)
-	l.auth.AuthenticationSucceeded(user, then, w, req)
+	glog.V(4).Infof(`Login with provider %q succeeded for %q: %#v`, l.provider, username, authResponse)
```
Comment (enj) [inline pkg/oauthserver/server/login/login.go]: log authResponse.User

File: pkg/oauthserver/authenticator/interfaces.go (hunk 4 above)
Comment (deads2k) [reply pkg/oauthserver/authenticator/interfaces.go]: > both of these should take a context now I suppose

They can, but you'll want to consider that separately.

File: pkg/oauthserver/authenticator/password/allowanypassword/anyauthpassword.go (hunk 2 above)
Comment (deads2k) [reply pkg/oauthserver/authenticator/password/allowanypassword/anyauthpassword.go]: > Please update `identitymapper.UserFor` to return `authenticator.Response`

You can do that as a followup.  Think carefully about whether you think an identity should be able to choose an audience.  I think that's probably a mistake.

**Comment by deads2k** [comment]
green, blocks rebase, merging

**Pull request merged**